auto3dseg/algorithm_templates/dints/configs/hyper_parameters.yaml

```diff
@@ -1,61 +1,61 @@
 ---
-# hyper-parameters
-amp: true
-bundle_root: null
-ckpt_path: "$@bundle_root + '/model_fold' + str(@fold)"
-data_file_base_dir: null
-data_list_file_path: null
-determ: false
-fold: 0
-input_channels: null
-learning_rate: 0.2
-num_images_per_batch: 2
-num_iterations: 40000
-num_iterations_per_validation: 500
-num_patches_per_image: 1
-num_sw_batch_size: 2
-output_classes: null
-overlap_ratio: 0.625
-patch_size: null
-patch_size_valid: null
-softmax: true
+training:
+  # hyper-parameters
+  amp: true
+  bundle_root: null
+  ckpt_path: "$@training#bundle_root + '/model_fold' + str(@training#fold)"
+  data_file_base_dir: null
+  data_list_file_path: null
+  determ: false
+  fold: 0
+  input_channels: null
+  learning_rate: 0.2
+  num_images_per_batch: 2
+  num_iterations: 40000
+  num_iterations_per_validation: 500
+  num_patches_per_image: 1
+  num_sw_batch_size: 2
+  output_classes: null
+  overlap_ratio: 0.625
+  patch_size: null
+  patch_size_valid: null
+  softmax: true
 
-# training
-loss:
-  _target_: DiceFocalLoss
-  include_background: true
-  to_onehot_y: "$@softmax"
-  softmax: "$@softmax"
-  sigmoid: "$not @softmax"
-  squared_pred: true
-  batch: true
-  smooth_nr: 1.0e-05
-  smooth_dr: 1.0e-05
-optimizer:
-  _target_: torch.optim.SGD
-  lr: "@learning_rate"
-  momentum: 0.9
-  weight_decay: 4.0e-05
-lr_scheduler:
-  _target_: torch.optim.lr_scheduler.StepLR
-  optimizer: "$@optimizer"
-  step_size: "$@num_iterations // 5"
-  gamma: 0.5
+  loss:
+    _target_: DiceFocalLoss
+    include_background: true
+    to_onehot_y: "$@training#softmax"
+    softmax: "$@training#softmax"
+    sigmoid: "$not @training#softmax"
+    squared_pred: true
+    batch: true
+    smooth_nr: 1.0e-05
+    smooth_dr: 1.0e-05
+  optimizer:
+    _target_: torch.optim.SGD
+    lr: "@training#learning_rate"
+    momentum: 0.9
+    weight_decay: 4.0e-05
+  lr_scheduler:
+    _target_: torch.optim.lr_scheduler.StepLR
+    optimizer: "$@training#optimizer"
+    step_size: "$@training#num_iterations // 5"
+    gamma: 0.5
 
 # fine-tuning
 finetune:
   activate: false
-  pretrained_ckpt_name: "$@bundle_root + '/model_fold' + str(@fold) + '/best_metric_model.pt'"
+  pretrained_ckpt_name: "$@training#bundle_root + '/model_fold' + str(@training#fold) + '/best_metric_model.pt'"
```

Comment:
Just to confirm, the folder structure will be something like:
```bash
└── dints0
    ├── arch_ram_fold0
    ├── configs
    ├── model_fold0
    └── scripts
└── dints1
    ├── arch_ram_fold1
    ├── configs
    ├── model_fold1
    └── scripts
```
It is fine by me

Reply:
yes. it is expected.

Reply:
Can we one bundle_root for training and searching, if it is expected to share the same folder?

Reply:
Another benefit of keeping one bundle_root for both training and searching: the user can use the following syntax for dints and other algorithm to run the bundle

```bash
torchrun --nnodes=1 \
    --nproc_per_node=${N_GPU} -m \
    scripts.train run \
    --config_file ...
    --bundle_root ${FOLDER_PATH} "
```

Reply:
that's fair.

Reply:
just update it.